### ISSUE TITLE
rename(2) syscall, same as link(2)

### ISSUE THREAD

**Issue by DoctorWkt** [eschaton/MINIXCompat#19]
Chris, the functions to do `rename(2)` are pretty much the same as for `link(2)`. I've got the code here but my commit is on top of all my other changes, so it won't apply to your main branch at the moment :-)

```
/*! MINIX `rename(2)` implementation. */
minix_syscall_result_t MINIXCompat_SysCall_rename(minix_syscall_func_t func, uint16_t src_dest, m68k_address_t msg, minix_message_t *message, uint32_t * _Nonnull out_result)
{
    // rename(2) sends mess1
    // - m3_i1: len(name)
    // - m3_i2: len(name2)
    // - m3_p1: name
    // - m3_p2: name2

    MINIXCompat_Message_Swap(mess1, message);
    int16_t minix_name_len = message->m1_i1;
    int16_t minix_name2_len = message->m1_i2;
    m68k_address_t minix_name = message->m1_p1;
    m68k_address_t minix_name2 = message->m1_p2;

    // Copy the names (paths) to host memory.
    // The name will include any trailing '\0' character. (Use of the pointer from the message rather than the message's inline copy is intentional, since the latter can only handle up to 14 characters.)

    char *minix_name_on_host = MINIXCompat_RAM_Copy_Block_To_Host(minix_name, minix_name_len);
    char *minix_name2_on_host = MINIXCompat_RAM_Copy_Block_To_Host(minix_name2, minix_name2_len);

    // Rename the file at that path, relative to MINIXCOMPAT_DIR.

    int16_t rename_err = MINIXCompat_File_Rename(minix_name_on_host, minix_name2_on_host);

    // rename(2) responds with mess1
    // - mess1.m_type: result

    MINIXCompat_Message_Clear(message);
    message->m_type = rename_err;

    MINIXCompat_Message_Swap(mess1, message);

    *out_result = rename_err;

    // Clean up.

    free(minix_name_on_host);
    free(minix_name2_on_host);

    return minix_syscall_result_success;
}
```

and

```
int16_t MINIXCompat_File_Rename(const char *minix_path, const char *minix_path2)
{
    int16_t result;

    assert(minix_path != NULL);
    assert(minix_path2 != NULL);

    char *host_path = MINIXCompat_Filesystem_CopyHostPathForPath(minix_path);
    char *host_path2 = MINIXCompat_Filesystem_CopyHostPathForPath(minix_path2);

    int rename_err = rename(host_path, host_path2);
    if (rename_err == 0) {
        result = 0;
    } else {
        result = -MINIXCompat_Errors_MINIXErrorForHostError(errno);
    }

    free(host_path);
    free(host_path2);

    return result;
}
```

**Comment by eschaton** [comment]
Thanks! Closing this as addressed by #17. :)

**Comment by eschaton** [comment]
Thanks! Closed as addressed with #17.
